<launch>
  <arg name="camera_id" default="2" /> <!-- 0 for main camera, 1 for tracking camera and 2 for webcames. -->
  <arg name="frame_rate" default="30" />
  <arg name="width" default="640" />
  <arg name="height" default="480" />
  <arg name="input_format" default="YUY2" /> <!-- format for the input stream from camera -->
  <arg name="output_format" default="RGB" /> <!-- format for the output stream -->
  <arg name="topic_name" default="arducam/image_raw" />
  <arg name="image_compress" default="true" /> 

  <node pkg="rb5_vision" type="rb_lepton" name="rb_arducam" output="screen">
    <param name="camera_id" value="$(arg camera_id)" />
    <param name="frame_rate" value="$(arg frame_rate)" />
    <param name="width" value="$(arg width)" />
    <param name="height" value="$(arg height)" />
    <param name="input_format" value="$(arg input_format)" />
    <param name="output_format" value="$(arg output_format)" />
    <param name="topic_name" value="$(arg topic_name)" />
    <param name="image_compress" value="$(arg image_compress)" />
  </node>
</launch>
